### PR TITLE
OS-Link button

### PR DESCRIPTION
**Pull request by slee1130**
### 🛠 Purpose

---

### ✏️ Notes to Reviewer
Changed the branch name, prefix with OS and made sure committed the right change I've made.
The rest will be the same as this morning: 
Clean up the codes
Rendering <a> tag to open a new tab 
Added horizontal rule to make the buttons separated
Added styling on Button.stories.styles.js

---

### 🖥 Screenshots

---

### CODE REVIEW

File: packages/Button/src/Button.styles.js [hunk 1]
```diff
@@ -20,7 +20,7 @@ const commonStyles = `
   font-weight: bold;
   text-align: center;
   vertical-align: middle;
-
+  text-decoration: none;
```
Comment (oscarkwan) [inline packages/Button/src/Button.styles.js]: add an empty line after

File: packages/Button/src/LinkButton.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { ShirtSizes } from "@paprika/helpers/lib/customPropTypes";
+import Button from "./Button";
+import buttonStyles from "./Button.styles";
+
+const LinkPropTypes = {
+  children: PropTypes.node.isRequired,
+  kind: PropTypes.oneOf(Button.Kinds.ALL),
+  size: PropTypes.oneOf(ShirtSizes.MEDIUM),
+  href: PropTypes.string,
+  text: PropTypes.string,
```
Comment (oscarkwan) [inline packages/Button/src/LinkButton.js]: We should not need a `text` prop but just rely on `children` instead. Let me know if you have any questions

File: packages/Button/src/LinkButton.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  href: null,
+  text: "Link",
+  isOpenNewTab: true,
+};
+
+const LinkButton = React.forwardRef((props, ref) => {
+  const buttonProps = {
+    children: null,
+  };
+  console.log(props);
```
Comment (oscarkwan) [inline packages/Button/src/LinkButton.js]: Remove console.logs

File: packages/Button/src/LinkButton.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+        css={buttonStyles}
+        {...props}
+        {...buttonProps}
+        ref={ref}
+        href={props.href}
+        target="_blank"
+        rel="noopener noreferrer"
+      >
+        {props.text}
+        {buttonProps.children}
```
Comment (oscarkwan) [inline packages/Button/src/LinkButton.js]: can we use the button prop children for the text instead?

File: packages/Button/src/LinkButton.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  text: "Link",
+  isOpenNewTab: true,
+};
+
+const LinkButton = React.forwardRef((props, ref) => {
+  const buttonProps = {
+    children: null,
+  };
+  console.log(props);
+  if (props.isOpenNewTab) {
```
Comment (oscarkwan) [inline packages/Button/src/LinkButton.js]: I'd create a separate variable that can hold

```
target="_blank"
rel="noopener noreferrer"
```

based on if there is a `prop.isOpenNewTab` or not. Then we don't need to render two different `<a>`. We just render one.

For example
`<a {...isOpenNewTabProps} {...buttonProps} etc....>`

File: packages/Button/stories/Button.stories.styles.js [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { Story } from "storybook/assets/styles/common.styles";
 import tokens from "@paprika/tokens";
 
 // Common Button story styles
+// Added link button story styles
```
Comment (oscarkwan) [inline packages/Button/stories/Button.stories.styles.js]: We don't need this comment

File: packages/Button/stories/Button.stories.styles.js [hunk 7]
```diff
@@ -12,4 +13,11 @@ export const ButtonStory = styled(Story)`
       margin-left: ${tokens.space};
     }
   }
+  a,
```
Comment (oscarkwan) [inline packages/Button/stories/Button.stories.styles.js]: You just need to add `a, button, ... { margin-left: ${tokens.space}; }`. You don't need an additional block that has the same css rule.

File: packages/Button/stories/examples/Basic.js [hunk 8]
```diff
@@ -78,6 +80,11 @@ const ExampleStory = () => (
       </Button>
     </p>
     <Rule />
+    <p>
+      <Button.Link onClick={clickHandler} isOpenNewTab={false} kind="primary" href="/about" text="Link" />
+      <Button.Link onClick={clickHandler} kind="secondary" href="/about" text="Link that opens in new tab" />
```
Comment (oscarkwan) [inline packages/Button/stories/examples/Basic.js]: `/about` is not a valid link. Just use google

File: packages/Button/src/LinkButton.js (hunk 3 above)
Comment (slee1130) [reply packages/Button/src/LinkButton.js]: done

File: packages/Button/stories/Button.stories.styles.js (hunk 6 above)
Comment (slee1130) [reply packages/Button/stories/Button.stories.styles.js]: done

File: packages/Button/stories/Button.stories.styles.js (hunk 7 above)
Comment (slee1130) [reply packages/Button/stories/Button.stories.styles.js]: done

File: packages/Button/stories/examples/Basic.js (hunk 8 above)
Comment (slee1130) [reply packages/Button/stories/examples/Basic.js]: done

File: packages/Button/src/LinkButton.js (hunk 4 above)
Comment (slee1130) [reply packages/Button/src/LinkButton.js]: done

File: packages/Button/src/LinkButton.js (hunk 2 above)
Comment (slee1130) [reply packages/Button/src/LinkButton.js]: done

File: packages/Button/src/Button.styles.js (hunk 1 above)
Comment (slee1130) [reply packages/Button/src/Button.styles.js]: done

File: packages/Button/src/LinkButton.js (hunk 5 above)
Comment (slee1130) [reply packages/Button/src/LinkButton.js]: done

**Pull request merged**